### ISSUE TITLE
Lack of documentation regarding use of Opium with SSL

### ISSUE THREAD

**Issue by rosalogia** [rgrinberg/opium#230]
I would appreciate some further documentation on how to use the ssl function in the App module. Right now I'm passing it the paths to my cert and privkey respectively, but it doesn't seem to be having any effect. I read some of the code for the App module and it seems like the cert and key, if they were passed at all, are extracted into `_mode` [here](https://github.com/rgrinberg/opium/blob/ddb3416796f77d226e15dc666f239a8d8723ffe3/opium/src/app.ml#L11), but it doesn't seem like `_mode` is ever referenced after creation. Why might that be?

**Comment by anuragsoni** [comment]
The `_mode` should be removed. The httpaf based opium (version 0.19.0) and above do not support setting up a ssl server at the moment.

**Comment by rosalogia** [comment]
I see; thank you!

**Comment by rosalogia** [comment]
Out of curiosity, is there an issue open that addresses this? If you don't mind, I'd like to inquire further about the problem and what it would take to solve it.

**Comment by anuragsoni** [comment]
The issue tracking the original work on the httpaf port had some details about it https://github.com/rgrinberg/opium/issues/138#issue-548455807

TLDR: It'll be easy to support it once the upstream httpaf project adds support for this.
